src/Illuminate/Support/Number.php

```diff
@@ -24,6 +24,45 @@ class Number
      */
     protected static $currency = 'USD';
 
+    /**
+     * Divides a number into pairs of specified intervals with optional delimiters and padding.
+     *
+     * @param  string|int|float  $number
+     * @param  int  $interval
+     * @param  string  $delimiter
+     * @param  string  $padding
+     * @return string
+     */
+    public static function group($number, int $interval = 1, string $delimiter = '', string $padding = '')
+    {
+        // Convert number to string
+        $number = (string) $number;
+
+        // Handle decimal part separately
+        [$integer, $decimal] = explode('.', $number.'.0');
```

Comment:
Not all numbers use a dot as a decimal separator. As `$number` accepts a string, that can be a user provided string from a form.

I'd either accept the separator as a parameter, defaulting to a dot, or drop the support for decimal places all together (check if `$number` is a float and truncate it).

Reply:
And if one would want to parse the number in a localized way, NumberFormatter would provide such functionality. One would refrain from doing it oneself.